File: sdks/ruby/lib/hooksniff/environment.rb
# frozen_string_literal: true

module HookSniff
  class Environment
    def initialize(api_client)
      @api_client = api_client
    end

    def list
      @api_client.execute(method: :get, path: "/api/v1/environments")
    end

    def create(environment_in)
      @api_client.execute(method: :post, path: "/api/v1/environments", body: environment_in)
    end

    def get(environment_id)
      @api_client.execute(method: :get, path: "/api/v1/environments/#{environment_id}")
    end

    def update(environment_id, environment_patch)
      @api_client.execute(method: :put, path: "/api/v1/environments/#{environment_id}", body: environment_patch)
    end

    def delete(environment_id)
      @api_client.execute(method: :delete, path: "/api/v1/environments/#{environment_id}")
    end

    def list_variables(environment_id)
      @api_client.execute(method: :get, path: "/api/v1/environments/#{environment_id}/variables")
    end

    def get_variable(environment_id, variable_id)
      @api_client.execute(method: :get, path: "/api/v1/environments/#{environment_id}/variables/#{variable_id}")
    end

    def create_variable(environment_id, variable_in)
      @api_client.execute(method: :post, path: "/api/v1/environments/#{environment_id}/variables", body: variable_in)
    end

    def update_variable(environment_id, variable_id, variable_in)
      @api_client.execute(method: :put, path: "/api/v1/environments/#{environment_id}/variables/#{variable_id}", body: variable_in)
    end

    def delete_variable(environment_id, variable_id)
      @api_client.execute(method: :delete, path: "/api/v1/environments/#{environment_id}/variables/#{variable_id}")
    end

    def bulk_upsert_variables(environment_id, bulk_in)
      @api_client.execute(method: :post, path: "/api/v1/environments/#{environment_id}/variables/bulk", body: bulk_in)
    end
  end
end

File: sdks/csharp/HookSniff/Environment.cs
using System.Collections.Generic;
using System.Threading.Tasks;

namespace HookSniff
{
    public class Environment
    {
        private readonly HookSniffClient _client;

        public Environment(HookSniffClient client)
        {
            _client = client;
        }

        public Task<List<Dictionary<string, object>>> List()
        {
            return _client.HookSniffHttpClient.GetAsync<List<Dictionary<string, object>>>("/api/v1/environments");
        }

        public Task<Dictionary<string, object>> Create(Dictionary<string, object> environmentIn)
        {
            return _client.HookSniffHttpClient.PostAsync<Dictionary<string, object>>("/api/v1/environments", environmentIn);
        }

        public Task<Dictionary<string, object>> Get(string environmentId)
        {
            return _client.HookSniffHttpClient.GetAsync<Dictionary<string, object>>($"/api/v1/environments/{environmentId}");
        }

        public Task<Dictionary<string, object>> Update(string environmentId, Dictionary<string, object> environmentPatch)
        {
            return _client.HookSniffHttpClient.PutAsync<Dictionary<string, object>>($"/api/v1/environments/{environmentId}", environmentPatch);
        }

        public Task Delete(string environmentId)
        {
            return _client.HookSniffHttpClient.DeleteAsync($"/api/v1/environments/{environmentId}");
        }

        public Task<List<Dictionary<string, object>>> ListVariables(string environmentId)
        {
            return _client.HookSniffHttpClient.GetAsync<List<Dictionary<string, object>>>($"/api/v1/environments/{environmentId}/variables");
        }

        public Task<Dictionary<string, object>> GetVariable(string environmentId, string variableId)
        {
            return _client.HookSniffHttpClient.GetAsync<Dictionary<string, object>>($"/api/v1/environments/{environmentId}/variables/{variableId}");
        }

        public Task<Dictionary<string, object>> CreateVariable(string environmentId, Dictionary<string, object> variableIn)
        {
            return _client.HookSniffHttpClient.PostAsync<Dictionary<string, object>>($"/api/v1/environments/{environmentId}/variables", variableIn);
        }

        public Task<Dictionary<string, object>> UpdateVariable(string environmentId, string variableId, Dictionary<string, object> variableIn)
        {
            return _client.HookSniffHttpClient.PutAsync<Dictionary<string, object>>($"/api/v1/environments/{environmentId}/variables/{variableId}", variableIn);
        }

        public Task DeleteVariable(string environmentId, string variableId)
        {
            return _client.HookSniffHttpClient.DeleteAsync($"/api/v1/environments/{environmentId}/variables/{variableId}");
        }

        public Task<List<Dictionary<string, object>>> BulkUpsertVariables(string environmentId, Dictionary<string, object> bulkIn)
        {
            return _client.HookSniffHttpClient.PostAsync<List<Dictionary<string, object>>>($"/api/v1/environments/{environmentId}/variables/bulk", bulkIn);
        }
    }
}
